### PR TITLE
web app layers without tests

### PR DESCRIPTION
**Pull request by panqqq**

### CODE REVIEW

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @PostMapping
+    public ResponseEntity<?> addDepartment(@RequestBody Department department) {
+        Department d = departmentService.addDepartment(department);
+        URI uri = MvcUriComponentsBuilder.fromController(getClass())
+                .path("/{id}")
+                .build(d.getId());
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .header("Location", uri.toString())
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/controllers/DepartmentController.java:45]: Why you need to return 'Location' uri in Header ?

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return departmentService.getAllDepartments();
+    }
+
+    @GetMapping("{id}")
+    public Department getDepartmentById(@PathVariable Long id) {
+        return departmentService.getDepartment(id);
+    }
+
+    @PostMapping
+    public ResponseEntity<?> addDepartment(@RequestBody Department department) {
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/controllers/DepartmentController.java:39]: ResponseEntity<?> -> ResponseEntity<Department>

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@RequestMapping("departments")
+public class DepartmentController {
+    private final DepartmentService departmentService;
+
+    public DepartmentController(DepartmentService departmentService) {
+        this.departmentService = departmentService;
+    }
+
+    @GetMapping
+    public List<Department> getAll() {
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/controllers/DepartmentController.java:29]: Change to be consistent with others. For others you have ResponseEntity returned

File: src/main/java/com/endava/webapp/model/Department.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Objects;
+
+@Entity
+public class Department {
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Long id;
+    @NotNull(message = "The name should not be null")
+    @NotEmpty(message = "The name should not be empty")
+    @NotBlank(message = "The name should not be blank")
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/model/Department.java:19]: NotBlank = NotEmpy + NotNull?

File: src/main/java/com/endava/webapp/model/Department.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    @NotEmpty(message = "The location should not be empty")
+    @NotBlank(message = "The location should not be blank")
+    private String location;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/model/Department.java:31]: You can use Lombok and add annotations for Getter, Setter ...

**Review by aledimov — COMMENTED** [review]
Why you need dependency <artifactId>ojdbc8</artifactId> ?

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java (hunk 2 above)
Comment (panqqq) [reply src/main/java/com/endava/webapp/controllers/DepartmentController.java]: fixed

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java (hunk 3 above)
Comment (panqqq) [reply src/main/java/com/endava/webapp/controllers/DepartmentController.java]: fixed

File: src/main/java/com/endava/webapp/model/Department.java (hunk 5 above)
Comment (panqqq) [reply src/main/java/com/endava/webapp/model/Department.java]: added

File: src/main/java/com/endava/webapp/model/Department.java (hunk 4 above)
Comment (panqqq) [reply src/main/java/com/endava/webapp/model/Department.java:22]: fixed

File: src/main/java/com/endava/webapp/controllers/DepartmentController.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+
+@RestController
+@RequestMapping("departments")
+public class DepartmentController {
+    private final DepartmentService departmentService;
+    @Autowired
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/controllers/DepartmentController.java:24]: add a blank line

File: src/main/java/com/endava/webapp/services/EmployeeService.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    public Employee getEmployee(Long id) {
+        return employeeRepository.findById(id).orElseThrow(() -> new NoSuchElementException("The employee ID " + id + " was not found!"));
+    }
+
+    public Employee add(Employee employee) {
```
Comment (aledimov) [inline src/main/java/com/endava/webapp/services/EmployeeService.java:25]: Read about '@transactional' and where it should be applied

File: src/main/java/com/endava/webapp/model/Employee.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    @ManyToOne(cascade = CascadeType.PERSIST)
+    private Department department;
+
+    @NotBlank(message = "The email should not be blank")
+    @Column(unique = true)
+    private String email;
+
+    @NotBlank(message = "The phone_number should not be blank")
+    @Column(name = "phone_number", unique = true)
+    private String phoneNumber;
```
Comment (eldlit) [inline src/main/java/com/endava/webapp/model/Employee.java:46]: Please add spaces in between fields declaration

**Pull request merged**